client.go

```diff
@@ -357,7 +357,7 @@ func DefaultRetryPolicy(ctx context.Context, resp *http.Response, err error) (bo
 	// errors and may relate to outages on the server side. This will catch
 	// invalid response codes as well, like 0 and 999.
 	if resp.StatusCode == 0 || (resp.StatusCode >= 500 && resp.StatusCode != 501) {
-		return true, nil
+		return true, fmt.Errorf("unexpected HTTP status %s", resp.Status)
```

Comment:
This is likely a backwards compatibility breakage that we should not do.

Reply:
I took this as a less-severe breaking change, because I don't see a way to accomplish better propagation of errors without breaking the API.

Is there any suggestion you can make around how to accomplish this without the change? What's the evaluation criteria for a breaking change when it adds valuable functionality?

Reply:
@evanphx I guess looking at this comment further, whether it's a breaking change might be up to who you ask. I've seen similar changes make it in to Go as a non-breaking change (since code still compiles), but the argument could definitely be made that it's breaking to some degree.

Reply:
@evanphx I can revert this change back so as not to break your API. Would you see value in offering this as a retry policy function from the package, but under a different name?

Reply:
I'm now realizing after a rebase that I need to also address the new error handling introduced above. I'll create an retry policy function with error propagation, and undo my breaking changes to this one.

Reply:
@evanphx I've moved this out to its own function now, so it's no longer a breaking change.